runtime/main/src/main/java/org/qbicc/runtime/main/ObjectModel.java

```diff
@@ -28,7 +33,43 @@
     /**
      * Get the java.lang.Class instance for the type ID.
      */
-    public static native Class<?> get_class_from_type_id(CNative.type_id typeId);
+    public static native Class<?> get_class_from_type_id (CNative.type_id typeId, Stdint.uint8_t dimensions);
+
+    /**
+     * Creates java.lang.Class instance for array class
+     *
+     * @param componentClass leaf element of the array
+     * @param dimensions dimensions of the array
+     * @return instance of java.lang.Class
+     */
+    public static synchronized Class<?> get_or_create_class_for_refarray(Class<?> componentClass, Stdint.uint8_t dimensions) {
+        if (classMap == null) {
+            classMap = new HashMap<>();
```

Comment:
`HashMap` can be replaced with `ConcurrentHashMap` to avoid making this method `synchronized` but `ConcurrentHashMap` uses `Unsafe` APIs which are not yet supported in qbicc.

Reply:
We definitely do not want to use a map here at all. Each `Class` should contain a field for the array type of that `Class`, which is lazily initialized by this helper method; thus no synchronization is necessary (just a CAS operation after the array `Class` is constructed).